File: src/Javascript/array/1.两数之和.js
/*
 * @lc app=leetcode.cn id=1 lang=javascript
 *
 * [1] 两数之和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map(); // {need, index}
    for (let i = 0; i < nums.length; i++ ) {
        if (map.has(nums[i])) return [map.get(nums[i]), i];
        map.set(target - nums[i], i);
    }
};
// @lc code=end

File: Python/array/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python
#
# [1] 两数之和
#

# @lc code=start
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashMap = dict()
        for index, value in enumerate(nums):
            if value in hashMap:
                return [hashMap[value], index]
            else:
                hashMap[target - value] = index 
# @lc code=end

